tooling/pkg/dsl/yaml.go

```diff
@@ -659,6 +659,46 @@ func UnmarshalStreamYAML(value *yaml.Node) (*GeneralizedType, error) {
 	return t, nil
 }
 
+func UnmarshalMapYAML(value *yaml.Node) (*GeneralizedType, error) {
+	if value.Kind != yaml.MappingNode {
+		return nil, parseError(value, "a !map must be specified with fields `keys` and optionally `values`")
```

Comment:
I think both `keys` and `values` are mandatory fields. We did discuss keys defaulting to string, but it doesn't currently right? I think that is the way it should be.

Reply:
Ah yes, that error message is wrong. Will fix.